torch/csrc/jit/ir.cpp

```diff
@@ -568,159 +569,43 @@ Value* Value::setUniqueName(const std::string & name) {
-  }
-  auto inp = findInput(name);
-  const Argument & arg = inp.second;
-  if (!inp.first) {
-    return tensor_as<T>(arg.default_value.value());
-  }
-  Node *producer = inp.first->node();
-  if (producer->kind() != prim::Constant) return at::nullopt;
-  auto value = producer->t(attr::value);
-  return tensor_as<T>(std::move(value));
-}
-
-template at::optional<int64_t> Node::get(Symbol name);
-template at::optional<double> Node::get(Symbol name);
-template at::optional<at::Tensor> Node::get(Symbol name);
-template at::optional<std::vector<int64_t>> Node::get(Symbol name);
-
-at::optional<IValue> Node::get(Symbol name) {
-  // TODO (apaszke): remove once tracer and compiler stop emitting attributes
   if (hasAttribute(name)) {
     switch (kindOf(name)) {
       case AttributeKind::i:
-        return IValue{as_tensor(i(name))};
+        return IValue(i(name));
+      case AttributeKind::f:
+        return IValue(f(name));
       case AttributeKind::t:
-        return IValue{as_tensor(t(name))};
+        return IValue(t(name));
       case AttributeKind::is:
-        return IValue{as_tensor(is(name))};
+        return IValue(is(name));
       default:
         throw std::runtime_error("get() NYI");
     }
   }
-  auto inp = findInput(name);
-  const Argument & arg = inp.second;
-  if (!inp.first) {
-    return IValue{arg.default_value.value()};
-  }
-  Node * producer = inp.first->node();
-  if (producer->kind() != prim::Constant) return at::nullopt;
-  auto value = producer->t(attr::value);
-  return IValue{std::move(value)};
+  return toIValue(namedInput(name));
 }
 
-Value* Node::input(Symbol name) {
-  // TODO (apaszke): remove once tracer and compiler stop emitting attributes
-  if (hasAttribute(name)) {
-    switch (kindOf(name)) {
-      case AttributeKind::i:
-        return owningGraph()->insertConstant(i(name));
-      case AttributeKind::is:
-        return owningGraph()->insertConstant(is(name));
-      case AttributeKind::t:
-        return owningGraph()->insertConstant(t(name));
-      default:
-        throw std::runtime_error("getValue() NYI");
-    }
-  }
-  auto inp = findInput(name);
-  if (inp.first) return inp.first;
-  return owningGraph()->insertConstant(inp.second.default_value.value());
-}
-
-// XXX: the first coordinate can be a nullptr, which means that you should use
-// the default value for this arg, because it's optional and missing
-std::pair<Value*, const Argument&> Node::findInput(Symbol name) {
-  if (!schema_) {
-    findSchema();
+Value* Node::namedInput(Symbol name) const {
+  if(hasAttribute(name)) {
+    // XXX - const cast because this really should not be modifying graph
+    // and once we remove attributes it no longer will
+    Value* v = insertConstant(const_cast<Graph&>(*owningGraph()), get(name).value());
+    // XXX - insert point can be anywhere since modifying the graph is unexpected,
+    // so this is completely unsafe and needs to be gone as soon as possible.
```

Comment:
Surprisingly no, because that is what I tried. Then it gets split by autodiff across forward and backward. It then fails because graph executor outputs must be primitives.